From: Ben Hutchings <ben@decadent.org.uk>
Date: Sat, 6 Dec 2014 22:44:35 +0000
Subject: [PATCH 7/9] sctp: Fix double-free introduced by bad backport in
 2.6.32.62

One deletion was omitted from the backport of upstream commit c485658bae87
("net: sctp: fix skb leakage in COOKIE ECHO path of chunk->auth_chunk").

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
---
 net/sctp/sm_statefuns.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/net/sctp/sm_statefuns.c b/net/sctp/sm_statefuns.c
index ac98a1e..5ca1edb 100644
--- a/net/sctp/sm_statefuns.c
+++ b/net/sctp/sm_statefuns.c
@@ -747,7 +747,6 @@ sctp_disposition_t sctp_sf_do_5_1D_ce(const struct sctp_endpoint *ep,
 
 		/* Make sure that we and the peer are AUTH capable */
 		if (!sctp_auth_enable || !new_asoc->peer.auth_capable) {
-			kfree_skb(chunk->auth_chunk);
 			sctp_association_free(new_asoc);
 			return sctp_sf_pdiscard(ep, asoc, type, arg, commands);
 		}
